app/models/v2/test_result.rb

```diff
@@ -118,6 +118,10 @@ def security_guide_version
       attributes['security_guide__version'] || try(:security_guide)&.version
     end
 
+    def last_scanned
+      system.test_results.max_by(&:end_time).end_time
```

Comment:
N+1

Reply:
Also isn't this an own field? This does not even make sense.

Reply:
According to the [schema description](https://github.com/RedHatInsights/compliance-backend/blob/master/spec/api/v2/schemas/test_result.rb#L116), the `last_scanned` returns _the date when the System has been reported a Test Result for the last time._

This lists `test_results` for the assigned systems and takes the latest `end_time`. What does not make sense to you?

Reply:
How are the two things different in the context of the test result model? Also it's a very inefficient N+1 if you try to list test results. Just check how many SQL queries a single API request to list all test results of a report does with this change.